Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,305 @@
+                }
+            }
+
+            return schemaList;
+        }
+
+        private static List<JObject> ConvertStringToJObjectList(string schema, out List<string> errorList, bool validateAsDictionary)
+        {
+            List<JObject> schemaList = new List<JObject>();
+            errorList = new List<string>();
+
+            try
+            {
+                //There could be three possible cases:
+                //1. TreeName mapped to the ForgeTree in the dictionary and custom rules handle dictionary.
+                //2. There are only ForgeTree without matching forge tree name custom rules handle ForgeTree list.
+                Dictionary<string, ForgeTree> forgeTrees = JsonConvert.DeserializeObject<Dictionary<string, ForgeTree>>(schema);
+
+                if (validateAsDictionary)
+                {
+                    schemaList.Add(JObject.Parse(schema));
+                }
+                else
+                {
+                    foreach (KeyValuePair<string, ForgeTree> kvp in forgeTrees)
+                    {
+                        ForgeTree forgeTree = kvp.Value;
+
+                        if (forgeTree.Tree == null)
+                        {
+                            // Deserialize into Dictionary does not throw exception but will have null "Tree" property if schema is just a ForgeTree.
+                            // try to deserialize string to forge tree directly
+                            JsonConvert.DeserializeObject<ForgeTree>(schema);
+                            schemaList.Add(JObject.Parse(schema));
+                            break;
+                        }
+
+                        schemaList.Add(SerializeToJObject(forgeTree));
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                errorList.Add(e.Message);
+            }
+
+            return schemaList;
+        }
+
+        private static JObject SerializeToJObject(object forgeTree)
+        {
+            string stringSchema = JsonConvert.SerializeObject(
+                forgeTree,
+                new JsonSerializerSettings
+                {
+                    DefaultValueHandling = DefaultValueHandling.Ignore, // Prevent default values from getting added to serialized json schema.
+                    Converters = new List<JsonConverter> { new Newtonsoft.Json.Converters.StringEnumConverter() } // Use string enum values instead of numerical.
+                });
+
+            return JObject.Parse(stringSchema);
+        }
+
+        private static List<JObject> GetSchemaFromPath(string path, out List<string> errorList, bool validateAsDictionary)
+        {
+            errorList = new List<string>();
+
+            try
+            {
+                string schema = File.ReadAllText(path);
+
+                List<JObject> res = ConvertStringToJObjectList(schema, out List<string> convertError, validateAsDictionary);
```

Comment:
>convertError [](http://example.com/codeflow?start=88&length=12)

Can this just be out errorList instead of creating a separate convertError variable?